packages/eas-cli/src/build/metadata.ts

```diff
@@ -26,5 +27,8 @@ export function collectMetadata<T extends Platform>(
     sdkVersion: ctx.commandCtx.exp.sdkVersion,
     trackingContext: ctx.trackingCtx,
     distribution: ctx.buildProfile.distribution ?? DistributionType.STORE,
+    appName: ctx.commandCtx.exp.name,
```

Comment:
could we fall back to some other name if the user hasn't provided a name in their app.json? this way we're not forcing people to add a new field to app.json

Reply:
It seems that the `exp.name` property can't be optional (https://github.com/expo/expo-cli/blob/master/packages/config-types/src/ExpoConfig.ts#L10) 🤔 If not specified it falls back to the package name - https://github.com/expo/expo-cli/blob/master/packages/config/src/Config.ts#L476

I'd say we don't have an issue here. I'll let myself merge the PR, let me know if there's something I'm missing.